File: coding-problems/python/e-242-valid-anagram.py
# import collections
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        countS = Counter(s)
        countT = Counter(t)
        # countS = collections.Counter(s)
        # countT = collections.Counter(t)

        for char, count in countS.items():
            if countT.get(char) != count:
                return False

        return True

sol = Solution()
print(sol.isAnagram("anagram", "nagaram"))

File: coding-problems/ruby/e-242-valid-anagram.rb
def is_anagram(s, t)
    
    return false if s.length() != t.length()
    
    hashS = countLetters(s)
    hashT = countLetters(t)
    
    hashS.each do |letter, count|
        return false if hashT[letter] != count
    end
    
    return true
end

def countLetters(str)
    hash = Hash.new{ |h, k| h[k] = 0 }
    
    str.each_char do |letter|
        hash[letter] += 1
    end    
    
    return hash
end

puts is_anagram("anagram", "nagaram")